dbms/src/Disks/DiskLocal.cpp

```diff
@@ -95,6 +106,19 @@ void DiskLocal::createDirectories(const String & path)
     Poco::File(disk_path + path).createDirectories();
 }
 
+void DiskLocal::clearDirectory(const String & path)
+{
+    std::vector<Poco::File> files;
+    Poco::File(disk_path + path).list(files);
+    for (auto & file : files)
+        file.remove(true);
```

Comment:
Extremely dangerous.

Let's clean only files on top level and throw exception if we try to remove directory
(this will be done automatically)

Reply:
Done.